lib/types/thermostat.js

```diff
@@ -0,0 +1,345 @@
+var inherits = require('util').inherits;
+
+module.exports = function (HAPnode, config, functions) {
+    var Accessory = HAPnode.Accessory;
+    var Service = HAPnode.Service;
+    var Characteristic = HAPnode.Characteristic;
+    var uuid = HAPnode.uuid;
+    var debug = HAPnode.debug;
+
+    var module = {};
+
+    module.newDevice = function (device, tempDisplayUnit) {
+        var temperatureDisplayUnit = tempDisplayUnit;
+        var Thermostat = {
+            device_url: function(request_type){
+                host = 'http://'+config.veraIP+':3480/data_request'
+                return host+'?id='+request_type+'&DeviceNum='+device.id
+            },
+
+            remoteRequest: function(url, params, callback){
+                debug("Requesting: %s", url);
+                //HAPnode.request.debug = true
+                return HAPnode.request({
+                    method:'GET',
+                    uri: url,
+                    qs: params,
+                    resolveWithFullResponse: true
+                }).then(callback.bind(this)).catch(function(e){
+                    debug(e.error);
+                    debug(e.options);
+                    debug(e.response);
+                });
+            },
+
+            getVeraVariable: function(serviceId, variable, callback){
+                var url = this.device_url('variableget');
+                var params = {serviceId: serviceId, Variable: variable}
+                return this.remoteRequest(url, params, callback);
+            },
+
+            executeVeraAction: function(params){
+                var url = this.device_url('action');
+                var callback = function(){};
+                return this.remoteRequest(url, params, callback);
+            },
+
+            getCurrentHeatingCoolingState: function(){
+                var serviceId = 'urn:upnp-org:serviceId:HVAC_UserOperatingMode1';
+                var variable = 'ModeStatus';
+                var callback = function (response){
+                    if (response.statusCode === 200){
+                        var vera_state = response.body;
+                        var state;
+                        switch(vera_state){
```

Comment:
@tzonto thanks for this awesome work! Another small piece of beta feedback: mode really only tells what mode it is currently in, not whether it is currently heating or cooling.  What might be more relevant is hvacstate which can be either: 'Idle', 'Heating', 'Cooling', 'FanOnly', 'PendingHeat', 'PendingCool', 'Vent'.  

**That said:** The Nest plugin for Vera was haphazardly built and does throw a different mode into the mix, "eco" which always maps to 'Idle.' They also decided to *not* include hvacstate.

So it would seem that hvacstate should be the primary trigger, that then falls back to this logic...

Thanks again for the great work, if I find time when the kid is down for a nap, ill do what I can to make a pull-request, but I figured I'd start a conversation to see if this had been considered before.